### DISCUSSION TITLE
HID service   Bluetooth keyboard and mouse

### DISCUSSION THREAD

**Discussion by bbhxwl** [nanoframework/Home#993] [Ideas]
### Description

Who knows C language can translate this library.

### How to solve the problem

Who knows C language can translate this library.

https://github.com/T-vK/ESP32-BLE-Keyboard
https://github.com/T-vK/ESP32-BLE-Mouse

**Comment by josesimoes** [comment]
This is not a bug. Switching to discussion to gather people feedback.

**Comment by AdrianSoundy** [comment]
With latest changes to Bluetooth this could be implemented in managed code.
Part of the spec for HID service is the Battery service which has already been done, see Bluetooth sample 3

The github references above only refer to the device extension to the HidDevice in 
https://github.com/espressif/arduino-esp32/tree/master/libraries/BLE
